readHR.py

```diff
@@ -0,0 +1,124 @@
+class EcgReader:
+
+    def __init__(self, file):
+        """
+
+        :param file: Input file for ECG reading and calculations
+
+        """
+        self.file = file
+        self.csv_check = True
+        self.data_type = True
+        self.time = []
+        self.voltage = []
+        self.peak_vector = []
+        self.filecheck()
+        if self.csv_check:
+            self.readfile()
+            self.datacheck()
+        if self.data_type and self.csv_check:
+            self.findrange()
+
+    def filecheck(self):
+        """
+            This method takes in ECG data from a CSV file inputted by the
+            user, and checks that it is the correct file format.
+
+            :return: If the file is not a CSV, the function will raise
+            "Error: File is not a .csv". For CSV files, the initialization
+            will move on to the next step.
+
+        """
+        if self.file.endswith('.csv'):
+            self.csv_check = True
+        else:
+            self.csv_check = False
+            self.time = None
+            self.voltage = None
+            print("Error: File is not a .csv")
+
+    def readfile(self):
+        """
+        This method reads the contents of the csv file and attempts to convert
+        numeric strings into floats before adding to time and voltage vectors.
+
+        :return: Time and voltage vectors with each entry from columns in the
+        original csv file.
+        """
+        import csv
+        import numpy as np
+        if self.file.endswith('.csv'):
+            with open(self.file) as ecg_Data_File:
+                ecg_reader = csv.reader(ecg_Data_File)
+                next(ecg_reader)
+                for row in ecg_reader:
+                    try:
+                        u = float(row[0])
+                        self.time.append(u)
+                    except:
+                        self.time.append(row[0])
```

Comment:
this would be thrown when it's not possible to cast the value to float, correct? In that case, I imagine you may not want to append the data to your vector and instead perform some other action? (Perhaps appending a NaN, which could be a bit dangerous or you could just let the exception be thrown)

Reply:
ah I see you're doing a data check later...perhaps a better structure would be to call `datacheck` from `readfile` so that you never end up in an intermediate state where `readfile` has run but `datacheck` has not. This could happen if users are using these methods after instantiating an object.

Reply:
you may also want an `is_data_valid` flag to indicate if the vectors this class is holding are ready for use